modules/apigee_edge_apidocs/src/Entity/ApiDoc.php

```diff
@@ -214,24 +267,38 @@ public static function baseFieldDefinitions(EntityTypeInterface $entity_type) {
       ->setDisplayConfigurable('form', TRUE)
       ->setDisplayConfigurable('view', TRUE);
 
-    $fields['status'] = BaseFieldDefinition::create('boolean')
+    $fields['status']
       ->setLabel(t('Publishing status'))
       ->setDescription(t('A boolean indicating whether the API Doc is published.'))
-      ->setDefaultValue(TRUE)
       ->setDisplayOptions('form', [
         'type' => 'boolean_checkbox',
         'weight' => 1,
       ]);
 
     $fields['created'] = BaseFieldDefinition::create('created')
       ->setLabel(t('Created'))
-      ->setDescription(t('The time that the entity was created.'));
+      ->setDescription(t('The time that the entity was created.'))
+      ->setRevisionable(TRUE);
 
     $fields['changed'] = BaseFieldDefinition::create('changed')
       ->setLabel(t('Changed'))
-      ->setDescription(t('The time that the entity was last edited.'));
+      ->setDescription(t('The time that the entity was last edited.'))
+      ->setRevisionable(TRUE);
 
     return $fields;
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  protected function urlRouteParameters($rel) {
+    $uri_route_parameters = parent::urlRouteParameters($rel);
+
+    if ($rel === 'revision-revert-form' && $this instanceof RevisionableInterface) {
```

Comment:
I think this is something that the entity route provider should handle.
https://github.com/fago/entity/blob/062c7c23900d0e5ef32060c2500143fcdeecbff5/src/Routing/RevisionRouteProvider.php#L76

Reply:
The _RevisionRouteProvider_ does provide the route, but the actual URL parameters passed are defined in the core Entity class (https://git.drupalcode.org/project/drupal/blob/8.6.x/core/lib/Drupal/Core/Entity/Entity.php#L314), and that doesn't pass the _apidoc_revision_ parameter for _'revision-revert-form',_ hence the need to add it here. Otherwise it throws this error:

`Symfony\Component\Routing\Exception\MissingMandatoryParametersException: Some mandatory parameters are missing ("apidoc_revision") to generate a URL for route "entity.apidoc.revision_revert_form". in Drupal\Core\Routing\UrlGenerator->doGenerate() (line 182 of /var/www/html/apigee_kickstarter_d8/web/core/lib/Drupal/Core/Routing/UrlGenerator.php).`